src/libraries/Microsoft.PowerFx.LanguageServerProtocol/LanguageServer/LanguageServer.cs

```diff
@@ -638,10 +639,10 @@ private void HandleRangeDocumentSemanticTokens(string id, string paramsJson)
                 return;
             }
 
-            HandleSemanticTokens(id, semanticTokensParams, TextDocumentNames.RangeDocumentSemanticTokens);
+            HandleSemanticTokens(id, semanticTokensParams, TextDocumentNames.RangeDocumentSemanticTokens, false);
         }
 
-        private void HandleSemanticTokens<T>(string id, T semanticTokensParams, string method)
+        private void HandleSemanticTokens<T>(string id, T semanticTokensParams, string method, bool isFullDocument = false)
```

Comment:
Why the need to pass this in? IsRangeSemanticTokens == false means it is full semantic tokens. We already check that in handleSemantictokens function. If you are trying to avoid string comparison method == TextDocumentNames.... then remove the isRangeSemanticTokens and use the negation of isFullSemanticTokens

Reply:
Ah good point, lets stick with this bool. Ill change the string comparison that is already existing to the bool here.